tests/src/tests/component_test_utils.go

```diff
@@ -0,0 +1,228 @@
+package tests
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+
+	schema "github.com/devfile/api/pkg/apis/workspaces/v1alpha2"
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/yaml"
+)
+
+func AddVolume(numVols int) []schema.VolumeMount {
+	commandVols := make([]schema.VolumeMount, numVols)
+	for i := 0; i < numVols; i++ {
+		commandVols[i].Name = "Name_" + GetRandomString(5, false)
+		commandVols[i].Path = "/Path_" + GetRandomString(5, false)
+		LogMessage(fmt.Sprintf("   ....... Add Volume: %s", commandVols[i]))
+	}
+	return commandVols
+}
+
+func getSchemaComponent(components []schema.Component, name string) (*schema.Component, bool) {
+	found := false
+	var schemaComponent schema.Component
+	for _, component := range components {
+		if component.Name == name {
+			schemaComponent = component
+			found = true
+			break
+		}
+	}
+	return &schemaComponent, found
+}
+
+func (devfile *TestDevfile) AddComponent(componentType schema.ComponentType) string {
+
+	var components []schema.Component
+	index := 0
+	if devfile.SchemaDevFile.Components != nil {
+		// Commands already exist so expand the current slice
+		currentSize := len(devfile.SchemaDevFile.Components)
+		components = make([]schema.Component, currentSize+1)
+		for _, component := range devfile.SchemaDevFile.Components {
+			components[index] = component
+			index++
+		}
+	} else {
+		components = make([]schema.Component, 1)
+	}
+
+	generateComponent(&components[index], componentType)
+	devfile.SchemaDevFile.Components = components
```

Comment:
I am confused with this function.
any reason `generateComponent` requires `&components` instead of just returning a component?
any reason need another copy of `devfile.SchemaDevFile.Components`?

can we just do 
```
devfile.SchemaDevFile.Components = append(devfile.SchemaDevFile.Components, generateComponent( componentType))
```

Reply:
Does append increment the size of a slice?

Reply:
different from java array, you do not need to defined size for a slice in golang
you do not need to do `components = make([]schema.Component, currentSize+1)`, as long as you defined the slice as `var components []schema.Component`, you are free to use it.

Reply:
Done